packages/hooks/src/useDeepMemo.ts

```diff
@@ -15,11 +15,35 @@ import { isEqual } from "lodash";
 import { useRef } from "react";
 
 // Continues to return the same instance as long as deep equality is maintained.
-export default function useDeepMemo<T>(value: T): T {
+export default function useDeepMemo<T extends object | undefined>(value: T): T {
   const ref = useRef(value);
+  const deepEqualCopiesRef = useRef<WeakSet<object>>();
+
+  if (value == undefined) {
```

Comment:
```suggestion
  if (value == undefined || value === ref.current) {
```

This will shortcut any weak set lookups or `isEqual` calls.